test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/CharArrayReaderTest.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.scalajs.testsuite.javalib.io
+
+import java.io._
+
+import org.junit.Assert._
+import org.junit.{After, Test}
+
+// Adapted from Apache Harmony CharArrayReaderTest.java
+//   https://github.com/apache/harmony/blob/02970cb7227a335edd2c8457ebdde0195a735733/classlib/modules/luni/src/test/api/common/org/apache/harmony/luni/tests/java/io/CharArrayReaderTest.java
+class CharArrayReaderTest {
+  private[io] val hw: Array[Char] = Array('H', 'e', 'l', 'l', 'o', 'W', 'o', 'r', 'l', 'd')
+  private[io] var cr: CharArrayReader = null
+
+  /**
+    * @tests java.io.CharArrayReader#CharArrayReader(char[])
+    */
+  @throws[IOException]
+  @Test def test_Constructor$C(): Unit = {
+    cr = new CharArrayReader(hw)
+    assertTrue("Failed to create reader", cr.ready)
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#CharArrayReader(char[], int, int)
+    */
+  @throws[IOException]
+  @Test def test_Constructor$CII(): Unit = {
+    cr = new CharArrayReader(hw, 5, 5)
+    assertTrue("Failed to create reader", cr.ready)
+    val c = cr.read
+    assertTrue("Created incorrect reader--returned '" + c.toChar + "' intsead of 'W'", c == 'W')
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#close()
+    */
+  @Test def test_close(): Unit = {
+    cr = new CharArrayReader(hw)
+    cr.close()
+    try {
+      cr.read
+      fail("Failed to throw exception on read from closed stream")
+    } catch {
+      case e: IOException =>
+
+      // Expected
+    }
+    // No-op
+    cr.close()
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#mark(int)
+    */
+  @throws[IOException]
+  @Test def test_markI(): Unit = {
+    cr = new CharArrayReader(hw)
+    cr.skip(5L)
+    cr.mark(100)
```

Comment:
Please also test cases with limits smaller than what is actually requested in `reset()`, so that we can make sure that it is indeed ignored, as the JavaDoc says.

Reply:
added